libflux/go/libflux/analyze_test.go

```diff
@@ -56,3 +58,170 @@ func TestAnalyze(t *testing.T) {
 		})
 	}
 }
+
+func TestMergePackages(t *testing.T) {
+	pkg1 := libflux.Parse(`
+package foo
+
+a = 1`)
+	pkg2 := libflux.Parse(`
+package foo
+
+b = 1`)
+	err := libflux.MergePackages(pkg1, pkg2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	gotFB, err := pkg2.MarshalFB()
+	got := ast.DeserializeFromFlatBuffer(gotFB)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := &ast.Package{
+		Package: "foo",
+		Files: []*ast.File{
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID:   &ast.Identifier{Name: "b"},
+						Init: &ast.IntegerLiteral{Value: 1},
+					},
+				},
+			},
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID:   &ast.Identifier{Name: "a"},
+						Init: &ast.IntegerLiteral{Value: 1},
+					},
+				},
+			},
+		},
+	}
+
+	if !cmp.Equal(got, want, asttest.IgnoreBaseNodeOptions...) {
+		t.Errorf(
+			"MergePackages unexpected packages -want/+got:\n %v",
+			cmp.Diff(want, got, asttest.IgnoreBaseNodeOptions...),
+		)
+	}
+}
+
+func TestNoInFiles(t *testing.T) {
+	pkg1 := libflux.Parse(`package foo`)
+	pkg2 := libflux.Parse(`
+package foo
+
+b = 1`)
+
+	err := libflux.MergePackages(pkg1, pkg2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	gotFB, err := pkg2.MarshalFB()
+	got := ast.DeserializeFromFlatBuffer(gotFB)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := &ast.Package{
+		Package: "foo",
+		Files: []*ast.File{
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
```

Comment:
I'm confused by how this test function is named, since there does seem to be some input files here.  I wonder if you parsed the empty string `""`, is that what you're trying to test?

Reply:
Well parsing the empty string actually throws an error on the rust side because that package would have no package clause. 

Perhaps these "empty packages" tests are obsolete now that I've eliminated the logic that I wrote for the `ParseDirToHandle` function in my last iteration.

Reply:
Yeah, I think I'm just gonna replace those two tests. They don't really make sense anymore.